RockLib.Configuration.ProxyFactory/ConfigurationProxyFactory.cs

```diff
@@ -0,0 +1,142 @@
+
+            var moduleBuilder = assemblyBuilder.DefineDynamicModule("MainModule");
+
+            var typeBuilder = moduleBuilder.DefineType(name, classTypeAttributes, typeof(object), interfaces);
+
+            var readonlyFields = new List<Tuple<string, FieldBuilder>>();
+
+            foreach (var property in type.GetTypeInfo().GetProperties())
+            {
+                const MethodAttributes methodAttributes = MethodAttributes.Public | MethodAttributes.HideBySig | MethodAttributes.SpecialName | MethodAttributes.Virtual;
+                var backingFieldName = "<" + property.Name + ">k__BackingField";
+
+                if (property.CanWrite)
+                {
+                    var fieldBuilder = typeBuilder.DefineField(backingFieldName, property.PropertyType, FieldAttributes.Private);
+                    var propertyBuilder = typeBuilder.DefineProperty(property.Name, PropertyAttributes.HasDefault, property.PropertyType, null);
+                    var getMethodBuilder = GetGetMethodBuilder(property.Name, property.PropertyType, typeBuilder, methodAttributes, fieldBuilder);
+                    var setMethodBuilder = GetSetMethodBuilder(property.Name, property.PropertyType, typeBuilder, methodAttributes, fieldBuilder);
+                    propertyBuilder.SetGetMethod(getMethodBuilder);
+                    propertyBuilder.SetSetMethod(setMethodBuilder);
+                }
+                else
+                {
+                    var fieldBuilder = typeBuilder.DefineField(backingFieldName, property.PropertyType, FieldAttributes.Private | FieldAttributes.InitOnly);
+                    var propertyBuilder = typeBuilder.DefineProperty(property.Name, PropertyAttributes.None, property.PropertyType, null);
+                    var getMethodBuilder = GetGetMethodBuilder(property.Name, property.PropertyType, typeBuilder, methodAttributes, fieldBuilder);
+                    propertyBuilder.SetGetMethod(getMethodBuilder);
+                    readonlyFields.Add(Tuple.Create(property.Name, fieldBuilder));
+                }
+            }
+
+            var constructorBuilder = typeBuilder.DefineConstructor(MethodAttributes.Public, CallingConventions.Standard, readonlyFields.Select(f => f.Item2.FieldType).ToArray());
+            var il = constructorBuilder.GetILGenerator();
+            for (int i = 0; i < readonlyFields.Count; i++)
+            {
+                constructorBuilder.DefineParameter(i + 1, ParameterAttributes.None, readonlyFields[i].Item1);
+                il.Emit(OpCodes.Ldarg_0);
+                il.Emit(OpCodes.Ldarg, i + 1);
+                il.Emit(OpCodes.Stfld, readonlyFields[i].Item2);
+            }
+
+            il.Emit(OpCodes.Ret);
+
+            return typeBuilder.CreateTypeInfo().AsType();
+        }
+
+        private static MethodBuilder GetGetMethodBuilder(string name, Type type,
+            TypeBuilder tb, MethodAttributes methodAttributes, FieldBuilder fieldBuilder)
```

Comment:
Should tb be renamed to typeBuilder, to match the rest of the naming?  Same for GetSetMethodBuilder param.